packages/hoppscotch-sh-admin/src/composables/useConfigHandler.ts

```diff
@@ -212,8 +149,57 @@ export function useConfigHandler(updatedConfigs?: ServerConfigs) {
     );
   };
 
+  /**
+   * The updated configs are transformed into a format that can be used by the mutations
+   */
+
+  const workingUpdatedConfigs: ConfigTransform[] = [
+    {
+      config: GOOGLE_CONFIGS,
+      enabled: updatedConfigs?.providers.google.enabled,
+      field: updatedConfigs?.providers.google.fields,
+    },
+    {
+      config: GITHUB_CONFIGS,
+      enabled: updatedConfigs?.providers.github.enabled,
+      field: updatedConfigs?.providers.github.fields,
+    },
+    {
+      config: MICROSOFT_CONFIGS,
+      enabled: updatedConfigs?.providers.microsoft.enabled,
+      field: updatedConfigs?.providers.microsoft.fields,
+    },
+    {
+      config: MAIL_CONFIGS,
+      enabled: updatedConfigs?.mailConfigs.enabled,
+      field: updatedConfigs?.mailConfigs.fields,
+    },
+  ];
+
+  // Push or filter the configs based on the enabled condition
+  const transformInfraConfigs = (workingConfigs: ConfigTransform[]) => {
+    let newConfigs: UpdatedConfigs[] = [];
+
+    workingConfigs.forEach(({ config, enabled, field }) => {
+      config.forEach(({ name, key }) => {
+        if (enabled && field) {
+          const value = typeof field === 'string' ? field : String(field[key]);
+          newConfigs.push({ name, value });
+        } else {
+          newConfigs = newConfigs.filter((item) => item.name !== name);
+        }
+      });
+    });
+    newConfigs = newConfigs.filter((item) => item.name !== '');
+    return newConfigs;
+  };
+
   // Transforming the working configs back into the format required by the mutations
-  const updatedAllowedAuthProviders = computed(() => [
+  const updatedInfraConfigs = () =>
+    updatedConfigs ? transformInfraConfigs(workingUpdatedConfigs) : [];
```

Comment:
Is it necessary to expose the above function as a return value for the `useConfigHandler` composable? If not, it can be inlined within the `updateInfraCongifs()` function. We need not expose info that isn't consumed elsewhere in the current use case.